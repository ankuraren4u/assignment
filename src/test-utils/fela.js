import Enzyme, { shallow, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { createRenderer } from "fela";
import { RendererProvider, ThemeProvider } from "react-fela";

import { theme } from "./../theme";
Enzyme.configure({ adapter: new Adapter() });

const renderer = createRenderer();

export const felaShallow = node =>
  shallow(
    <RendererProvider renderer={renderer}>
      <ThemeProvider theme={theme}>{node}</ThemeProvider>
    </RendererProvider>
  );

export const felaMount = node =>
  mount(
    <RendererProvider renderer={renderer}>
      <ThemeProvider theme={theme}>{node}</ThemeProvider>
    </RendererProvider>
  );
