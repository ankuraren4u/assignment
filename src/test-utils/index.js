export { felaShallow, felaMount } from "./fela";
