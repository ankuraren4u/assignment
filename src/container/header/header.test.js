import { Header } from ".";
import { felaShallow } from "../../test-utils/";
describe("<Header />", () => {
  describe("By default", () => {
    let component;
    beforeAll(() => {
      component = felaShallow(<Header />);
    });

    it("renders", () => {
      expect(component.html()).toMatchSnapshot();
    });

    it("does nothing on clicks", () => {
      let clickEvent = () => {
        component.simulate("click");
      };
      expect(clickEvent).not.toThrow();
    });
  });
});
