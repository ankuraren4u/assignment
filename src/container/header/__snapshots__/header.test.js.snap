// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> By default renders 1`] = `"<div class=\\"ui-header a b c d e f g h i j\\"><div class=\\"ui-autosuggest b k l e m n\\"><input type=\\"text\\" value=\\"\\" class=\\"o p q c a r s\\" placeholder=\\"Search for Starwars Character\\"/></div></div>"`;
